distutils/cygwinccompiler.py

```diff
@@ -50,8 +50,7 @@
 import os
 import sys
 import copy
-from subprocess import Popen, PIPE, check_output
-import re
+from subprocess import Popen, check_output
 
 import distutils.version
 from distutils.unixccompiler import UnixCCompiler
```

Comment:
Some of the imports are unused and could be removed. But they are used in the tests, so that would need fixing.

I initially kept them here to make the diff smaller.

Reply:
I noticed that. I didn't remove them because it was used in tests. I think it would be better if it was removed and the corresponding tests is changed.

Reply:
Removed in 93e3790de8. It was indeed nicer to review with the smaller diff and then subsequently address this issue.